sdk/eventhub/event-hubs/CHANGELOG.md

```diff
@@ -1,15 +1,19 @@
 # Release History
 
-## 5.8.1 (Unreleased)
+## 5.9.0 (Unreleased)
 
 ### Features Added
 
 ### Breaking Changes
 
 ### Bugs Fixed
 
+- Fixing a bug where events were not always received in order [#23993](https://github.com/Azure/azure-sdk-for-js/issues/23993).
+
 ### Other Changes
 
+- The receiver no longer attempts to build batches of `maxMessageCount` size, instead, it returns batches as soon as they are received from the service, up to `maxMessageCount`.
```

Comment:
Addressed in https://github.com/Azure/azure-sdk-for-js/pull/24731/commits/998bf37a0e92c48e4fad8e57cc46aaa7af677b09

Reply:
Would "breaking changes" scare customers away?  If we never garantee returning max # of messages, I don't feel it is breaking.

Reply:
Yeah it depends if you're looking at it from documentary or behavioral point of view. We never promised the old behavior but there may be a few cases out there that took that old behavior as a requirement for whatever reason, and they could be surprised by the change.

Another option is to put the new behavior behind a flag, but I am not really enthusiastic about it because it is added complexity for the customer and for the library maintenance.

Pulling in the other architects: @joheredi, @mpodwysocki, @witemple-msft

/cc @xirzec

Reply:
That's a fair call-out, Jeremy.    To clarify, documented behavior in both SDK and product docs is that you'll receive between 0 and `maxMessageCount`.    While it is a change in behavior, I think that it's fair to look at the previous behavior as "worked by coincidence."

Reply:
I don't have a great understanding of what's going on here, but Is it fair to say the old behavior was a bug? If I compare this to other JS APIs like streams, you may have a max chunk size but that doesn't mean the stream will wait for a chunk of that size before pushing it. If the events are very slow to receive, doesn't the method that we were using previously end up making the user wait a long time for data just because it is slow rolling?

If we never promised the previous chunking behavior and it was possible for us to get blocks of messages with fewer than `maxMessageCount` items, then I don't think this is much of a breaking change. What client code could have been written that would break?

Reply:
Reverted the change in https://github.com/Azure/azure-sdk-for-js/pull/24731/commits/990b71d943aa1b8159138a7ae41b3722d5afb37e.

Reply:
@witemple-msft previously we would wait until a max wait time then return whatever we receive so far up to the maxMessageCount.  After this PR IIUC we return received messages more frequently without wait until max wait time elapse.

Reply:
>  previously we would wait until a max wait time
we also return as soon as we already receive maxMessageCount of messages

Reply:
Discussed this with @deyaaeldeen and I think this is not a breaking change -- not even behaviorally. In terms of programmer assumptions, what we're changing is that before the change we would _either_ return `maxMessageCount` messages _or_ wait `maxWaitTimeInSeconds` time to return. We could return anywhere from 0-`maxMessageCount` messages upon return, but we'd only return fewer in case we waited the full wait time.

We never documented or guaranteed this exact behavior as far as I can tell, and I can't envision how a dependency on that assumption would be encoded into a program's logic. We are still returning between 0 and `maxMessageCount` messages, and client code always had to handle this situation, so I don't think it's breaking in any meaningful way. Maybe I'm wrong and someone out there has a program that is super sensitive to this assumption, but I've been thinking about it for a while and cannot really imagine that being the case.